core/src/validator.rs

```diff
@@ -744,25 +744,23 @@ impl Validator {
                 .register_exit(Box::new(move || cancel_tpu_client_next.cancel()));
         }
 
-        let accounts_update_notifier = geyser_plugin_service
-            .as_ref()
-            .and_then(|geyser_plugin_service| geyser_plugin_service.get_accounts_update_notifier());
-
-        let transaction_notifier = geyser_plugin_service
-            .as_ref()
-            .and_then(|geyser_plugin_service| geyser_plugin_service.get_transaction_notifier());
-
-        let entry_notifier = geyser_plugin_service
-            .as_ref()
-            .and_then(|geyser_plugin_service| geyser_plugin_service.get_entry_notifier());
-
-        let block_metadata_notifier = geyser_plugin_service
-            .as_ref()
-            .and_then(|geyser_plugin_service| geyser_plugin_service.get_block_metadata_notifier());
-
-        let slot_status_notifier = geyser_plugin_service
-            .as_ref()
-            .and_then(|geyser_plugin_service| geyser_plugin_service.get_slot_status_notifier());
+        let (
+            accounts_update_notifier,
+            transaction_notifier,
+            entry_notifier,
+            block_metadata_notifier,
+            slot_status_notifier,
+        ) = if let Some(service) = &geyser_plugin_service {
+                (
+                    geyser_plugin_service.get_accounts_update_notifier(),
+                    geyser_plugin_service.get_transaction_notifier(),
+                    geyser_plugin_service.get_entry_notifier(),
+                    geyser_plugin_service.get_block_metadata_notifier(),
+                    geyser_plugin_service.get_slot_status_notifier(),
```

Comment:
CI failed because you're still referencing `geyser_plugin_service` which is an `Option<_>`; since we unwrap the option in the `if let Some(...)` as `service`, these lines need to be `service.get_*()`